packages/plugin-manifest-validator/src/index.ts

```diff
@@ -18,60 +18,136 @@ interface SchemaValidateFunction {
   errors?: Array<Partial<ErrorObject>>;
 }
 
+type ValidatorResult =
+  | boolean
+  | { valid: true }
+  | { valid: false; message?: string };
+
+type Options = {
+  relativePath?: (filePath: string) => boolean;
+  maxFileSize?: (maxBytes: number, filePath: string) => ValidatorResult;
+  fileExists?: (filePath: string) => ValidatorResult;
+};
+
 /**
  * @param {Object} json
  * @param {Object=} options
  * @return {{valid: boolean, errors: Array<!Object>}} errors is null if valid
  */
 export default (
   json: Record<string, any>,
-  options: { [s: string]: (...args: any) => boolean } = {}
+  options: Options = {}
 ): ValidateResult => {
-  let relativePath = (...args: any) => true;
-  let maxFileSize = (...args: any) => true;
+  let relativePath: Options["relativePath"] = () => true;
+  let maxFileSize: Options["maxFileSize"] = () => true;
+  let fileExists: Options["fileExists"] = () => true;
+
   if (typeof options.relativePath === "function") {
-    relativePath = options.relativePath;
+    relativePath = options.relativePath as Options["relativePath"];
   }
   if (typeof options.maxFileSize === "function") {
-    maxFileSize = options.maxFileSize;
+    maxFileSize = options.maxFileSize as Options["maxFileSize"];
+  }
+  if (typeof options.fileExists === "function") {
+    fileExists = options.fileExists as Options["fileExists"];
   }
 
   const ajv = new Ajv({
     allErrors: true,
     formats: {
       "http-url": (str: string) => validateUrl(str, true),
       "https-url": (str: string) => validateUrl(str),
-      "relative-path": relativePath,
+      "relative-path": relativePath as () => boolean,
     },
   });
 
   const validateMaxFileSize: SchemaValidateFunction = (
     schema: string,
-    data: string
+    filePath: string
   ) => {
     // schema: max file size like "512KB" or 123 (in bytes)
     // data: path to the file
+    if (maxFileSize === undefined) {
+      return true;
+    }
+
     const maxBytes = bytes.parse(schema);
-    const valid = maxFileSize(maxBytes, data);
-    if (!valid) {
+    const result = maxFileSize(maxBytes, filePath);
+    const defaultMessage = `file size should be <= ${schema}`;
+
+    if (result === false) {
+      validateMaxFileSize.errors = [
+        {
+          keyword: "maxFileSize",
+          params: {
+            limit: maxBytes,
+          },
+          message: defaultMessage,
+        },
+      ];
+      return false;
+    }
+
+    if (typeof result === "object" && !result.valid) {
       validateMaxFileSize.errors = [
         {
           keyword: "maxFileSize",
           params: {
             limit: maxBytes,
           },
-          message: `file size should be <= ${schema}`,
+          message: result.message ?? defaultMessage,
         },
       ];
+      return false;
     }
-    return valid;
+
+    return true;
+  };
+
+  const validateFileExists: SchemaValidateFunction = (
+    schema: string,
```

Comment:
MUST: 
To achieve AC1.2. , we need to check `schema` value.
If we don't check `schema`, this validation function will run even if `fileExists: false`.
> 1.2. If the value is true, the manifest-validator will check file existence using a user-defined validator function 

NOTE: Currently, our JSON schema has no `fileExists: false`, so there is no immediate negative effect.

The `schema` type should be `boolean` but this will conflict with `SchemaValidateFunction`.
You can remove `SchemaValidateFunction` at line 14 and import `SchemaValidateFunction` from Ajv.

```suggestion
    schema: boolean,
```